vmm/src/sigsys_handler.rs

```diff
@@ -57,18 +112,28 @@ mod tests {
             libc::SYS_munmap,
             libc::SYS_rt_sigprocmask,
             libc::SYS_rt_sigreturn,
+            libc::SYS_sched_getaffinity,
             libc::SYS_set_tid_address,
             libc::SYS_sigaltstack,
+            libc::SYS_write,
         ];
+
         assert!(setup_seccomp(SeccompLevel::Basic(REQUIRED_SYSCALLS)).is_ok());
+        let sys_idx = libc::SYS_getpid as usize;
+        assert_eq!(METRICS.seccomp.bad_syscalls[sys_idx].count(), 0);
 
         // Calls the blacklisted SYS_getpid.
         let _pid = process::id();
 
-        // The signal handler should let the program continue.
-        assert!(true);
+        assert!(cpu_count() > 0);
 
-        // The reason this test doesn't check the failure metrics as well is that the signal handler
-        // doesn't work right with kcov - possibly because the process is being pinned to 1 core.
+        // We expect cpu_count to be 1 when kcov is testing coverage. When this changes, we'll have
```

Comment:
nit: this isn't immediately obvious, and it would be hard for anyone wanting to validate this info to actually do so. I think an explanation or a link would help. `kcov` doesn't document that it pins the process to 1 cpu, but there's a line in the changelog about it and there's of course the [code](https://github.com/SimonKagstrom/kcov/blob/cc4f205f8683aa55c0116075701c137d50f24a99/src/engines/ptrace_linux.cc#L408)

Reply:
For this one I think it might be worth opening an issue to cargo kcov or kcov to ask if there is a way to find out if a test is run as part of cargo test or cargo kcov. Maybe there's a fancy feature that we don't know about, or maybe they might even add something if more people find such a feature useful.